### ISSUE TITLE
appcache fallback (SPA) doesn't work

### ISSUE THREAD

**Issue by whitecolor** [NekR/offline-plugin#369]
I have `appShell` option configured:
```js
appShell: '/',
```

generated manifest:
```
CACHE MANIFEST
#ver:2018-4-15 02:02:43
#plugin:4.9.0

CACHE:
/bundles/18f4426789edc9a053f7.worker.js
/bundles/1003-1139-1140.lp.ru-8304ca6896f912a71a94.js
/bundles/1003-1139-1140-1288.lp.ru-158bd16363e2848602fc.js
/bundles/1003-1139-1140.geonames.ru-0db197ba553df3748f0d.js
/bundles/1003-1139-1140.geonames.en-1bcf43d7dc55642a1236.js
/bundles/1003.lp.ru-57a31de2577ade74cc76.js
/bundles/cities15000-8eeebdc8de3eee217bc0.js
/bundles/app-9bb92bf87ad5aed4cb67.js
/bundles/fonts-e129712248aa27593003.js
/bundles/vendor-33d2c45daef7f3aece84.js
/bundles/../

NETWORK:
*

FALLBACK:
/bundles/ /
```

In appcache `offline` mode path `site.com/` works, but any other paths like ``site.com/something` doesn't work. Service worker version works ok for such paths. Tested in Firefox and Chrome.

What can be the problem?

**Comment by NekR** [comment]
Well, issues is that FALLBACK is generated only to `/bundles/` subdirectory. Probably `appShell` does it wrong. Can you show me the metadata generated in ServiceWorker file? (json data in the beginning of the file), and your offline-plugin config if possible.

**Comment by whitecolor** [comment]
@NekR is there something wrong with meta?

**Comment by NekR** [comment]
@whitecolor Yeah, I meant that. But actually it doesn't have the information I was looking for, my bad. But the issues here is that you cache `/bundles/` URL, but set `appShell` to `/` URL, which isn't cached. Just set it to `appShell: '/bundles/'.

But I guess you have `/bundles/` specified as your `publicPath`, right? If so, then it's a bug because `appShell` option doesn't pick it up, as it should.

Let me know :-)

**Comment by whitecolor** [comment]
Yes this is the config for webpack output:
```
    path: join(outputPath!, 'bundles'),
    publicPath: '/bundles/'
```

**Comment by NekR** [comment]
@whitecolor thanks! Try to set `appShell: '/bundles/'`. I'm thinking about it now, if it's a bug or not. I'm not sure. `appShell` meant to receive an absolute path, so maybe not. At the very least, it should warn when supplied path is outside of ServiceWorker scope.

**Comment by NekR** [comment]
Well, I'll need public url of the website to test the things. Can't help by just guessing stuff.

**Comment by whitecolor** [comment]
Public is still served only over https, will it be possible to test appcache?

**Comment by NekR** [comment]
Yeah, I can do that in old Safari. (btw no need to serve things from http, https only is great)

**Comment by whitecolor** [comment]
https://charta.life

**Comment by whitecolor** [comment]
Just noticed that my HTML template doesn't have `manifest="appcache/manifest.appcache"` in `html`, though it doesn't really change things. If you don't mind a couple of questions regarding this:

1) why `manifest="appcache/manifest.appcache"` in html is inserted (if it works without it)?

2) Is it possible not to insert it in webpack-html-plugin auto generated HTML?
